metadata-proxy/nginx.conf

```diff
@@ -0,0 +1,63 @@
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+user www-data;
+worker_processes 4;
+pid /run/nginx.pid;
+
+events {
+  worker_connections 20;
+}
+
+http {
+  server {
+    listen 127.0.0.1:988;
```

Comment:
So I'll let you decide if it's worth it in this PR or not, but this is what you could do here:

1. Add variables for `PORT` and `METADATA_IP` (e.g. `__PORT__` and `__METADATA_IP__`)
2. Update the start-proxy.sh script to replace those variables in the config with the value from an environment variable, using sed, e.g. `sed -i s/__PORT__/${PORT:-988}/g /etc/nginx/nginx.conf`
3. Populate the env vars from the pod spec